x-pack/plugins/lens/public/shared_components/empty_placeholder.tsx

```diff
@@ -9,17 +9,22 @@ import React from 'react';
 import { EuiIcon, EuiText, IconType, EuiSpacer } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 
-export const EmptyPlaceholder = (props: { icon: IconType }) => (
+const noResultsMessage = (
+  <FormattedMessage id="xpack.lens.xyVisualization.noDataLabel" defaultMessage="No results found" />
+);
+
+export const EmptyPlaceholder = ({
+  icon,
+  message = noResultsMessage,
+}: {
+  icon: IconType;
+  message?: JSX.Element;
+}) => (
   <>
     <EuiText className="lnsChart__empty" textAlign="center" color="subdued" size="xs">
-      <EuiIcon type={props.icon} color="subdued" size="l" />
+      <EuiIcon type={icon} color="subdued" size="l" />
       <EuiSpacer size="s" />
-      <p>
-        <FormattedMessage
-          id="xpack.lens.xyVisualization.noDataLabel"
-          defaultMessage="No results found"
-        />
-      </p>
+      <p>{message}</p>
```

Comment:
Nitpick: would something like this be ok?
```jsx
{message ?? <FormattedMessage id="xpack.lens.xyVisualization.noDataLabel" defaultMessage="No results found" />}
```
I find it a bit more readable than the default assignment, scrolling up and down to find the fallback assignment.

But it's just my taste, so no strong opinion about it.

Reply:
missed this comment, I'll address it in gauges PR :)